Assignment5/GenericsHomwork.Tests/NodeTests.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using GenericsHomework;
+using System;
+
+namespace GenericsHomwork.Tests
+{
+    [TestClass]
+    public class NodeTests
+    {
+        [TestMethod]
+        public void Constructor_ValidInput_Success()
+        {
+            int number = 4;
+            Node<int> testNode = new Node<int>(number);
+            Assert.IsNotNull(testNode);
+        }
+        [ExpectedException(typeof(ArgumentNullException))]
+        [TestMethod]
+        public void Constructor_NullInput_Success()
+        {
+            int? number = null;
+            Node<int?> testNode = new Node<int?>(number);
+        }
+        [TestMethod]
+        public void Constructor_NullNext_Success()
+        {
+            int number = 4;
+            Node<int> testNode = new Node<int>(number, null);
+            Assert.IsNotNull(testNode);
+        }
+        [TestMethod]
+        public void Insert_ValidInput_Success()
+        {
+            int number = 4;
+            Node<int> testNode = new Node<int>(number);
+            testNode.Insert(number);
+            Assert.IsNotNull(testNode);
+        }
+        [ExpectedException(typeof(ArgumentNullException))]
+        [TestMethod]
+        public void Insert_NullInput_Success()
+        {
+            int? number = 4;
+            Node<int?> testNode = new Node<int?>(number);
+            number = null;
+            testNode.Insert(number);

```

Comment:
nit: My preference is to not reuse variables for different purposes. In this case since you really just have a single usage of each (constructor value and insert call), and the variable name is not adding much value (clarity of what the value means) you could just inline it.

NB: I also added the [null-forgiving operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving#:~:text=Available%20in%20C%23%208.0%20and%20later%2C%20the%20unary,prefix%20%21%20operator%20is%20the%20logical%20negation%20operator.) which would be needed to handle the warning that comes when you enable nullability.
```suggestion
            Node<int?> testNode = new Node<int?>(4);
            testNode.Insert(null!);
```